contracts/badger-ren/strategies/CurveSwapStrategy.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.8;
+
+import "deps/@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "deps/@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+import "deps/@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+import "deps/@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
+import "deps/@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+import "interfaces/curve/ICurveExchange.sol";
+
+contract CurveSwapStrategy is AccessControlUpgradeable, ReentrancyGuardUpgradeable {
+    using SafeMathUpgradeable for uint256;
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    // Only swapper is allowed to make swaps.
+    bytes32 public constant SWAPPER_ROLE = keccak256("SWAPPER_ROLE");
+    // The curve exchange contract is registered in the address provider with ID 2.
+    uint256 public constant CURVE_REGISTRY_EXCHANGE_ID = 2;
+
+    address public curveRegistryAddressProvider;
+
+    function initialize(address _admin, address _registry) public initializer {
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+
+        require(_admin != address(0x0), "must set admin address");
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+
+        require(_registry != address(0x0), "must set registry address provider address");
+        curveRegistryAddressProvider = _registry;
+    }
+
+    function swapTokens(
+        address _from,
+        address _to,
+        uint256 _amount,
+        uint256 _slippage
+    ) external nonReentrant onlySwapper returns (uint256 amount) {
+        (address registry, address pool, uint256 exchangeAmount) = _estimateSwapAmount(_from, _to, _amount);
+
+        _slippage = uint256(1e4).sub(_slippage);
```

Comment:
line 43 and 44 are same exact logic as seen in `_minAmount` on `BadgerRenAdapter`, perhaps consider moving it to common method?